### ISSUE TITLE
something wrong about the prediction

### ISSUE THREAD

**Issue by LIngerwsk** [LiamMaclean216/Pytorch-Transfomer#2]
hi,I think you have some problems in forecasting.
`if(output_sequence_length == 1):
        x[0].append([t(q).detach().squeeze().numpy()])
    else:
        for a in t(q).detach().squeeze().numpy():
            x[0].append([a])`
where the test data x, supposing the output_sequence_length is 5, then when you forecast the last 5 numbers, you shouldn't input the 5 Observed values, which is wrong.

**Comment by LiamMaclean216** [comment]
x isn't test data, it is the negative part of the sigmoid function. From this, the network is trying to predict the positive part.
